lib/values.rb

```diff
@@ -23,6 +23,11 @@ def self.with(hash)
           raise ArgumentError.new("Unexpected hash keys: #{unexpected_keys}")
         end
 
+        missing_keys = self::VALUE_ATTRS - hash.keys
+        if missing_keys.any?
+          raise ArgumentError.new("Missing hash keys: #{missing_keys}")
```

Comment:
would like this error to include the keys we got as well - more context is usually super helpful when dealing with most error messages.